rsa_extending_wiener_attack.sage

```diff
@@ -250,19 +250,17 @@ def example():
     from Crypto.Util.number import long_to_bytes, getPrime, bytes_to_long
     import uuid
     def rsa(e, n):
-        m = uuid.uuid4().hex.encode()
+        m = bytearray.fromhex("00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000045F22F0AE9FF2248CFA7E0884986D743BAB009BAEA71AC9C32658C307A95DFD2")
         c = pow(bytes_to_long(m), e, n)
-        return m.decode(), c
-    p = getPrime(1024)
```

Comment:
@X3NNY what are you supposed to do if you only know modulus, not `p` and `q`?

Reply:
Just throw the modulus N into the function `attack`.

The message format is just used to check the correctness of this attack.

Reply:
do you need to know P and Q?

Reply:
Definitely not.

One of the aim of this attack is to obtain the factorization of `N=pq`.

Reply:
Plus, the feasibility of RSA encryption is based on the difficulty of factorization of `N=pq`. If one have known the factorization of `N=pq`, he or she can already decrypt any ciphertext, so this is not related to the topic of this repo, namely `attack`, anymore.

Reply:
Isn't an RSA2048 modulus too big to factorize? I only know public key (modulus) and weak exponent (3). I don't know private exponent (d), or p or q.

Dead end, right?